db/compaction/compaction_job.cc

```diff
@@ -1256,8 +1254,9 @@ void CompactionJob::NotifyOnSubcompactionBegin(
   if (shutting_down_->load(std::memory_order_acquire)) {
     return;
   }
-  if (c->is_manual_compaction() && manual_compaction_paused_ &&
-      manual_compaction_paused_->load(std::memory_order_acquire) > 0) {
+  if (c->is_manual_compaction() &&
+      (manual_compaction_canceled_ &&
+       manual_compaction_canceled_->load(std::memory_order_acquire))) {
```

Comment:
nitpicking: it might be cleaner just make `manual_compaction_canceled_` a reference, then we don't need to worry if the pointer is valid or not:
```cpp
const std::atomic<bool>& manual_compaction_canceled_;
```

For auto-compaction, we can just pass a dummy atomic_bool which is always set to false.